Dropout.lua

```diff
@@ -20,7 +21,7 @@ function Dropout:updateOutput(input)
       self.output:resizeAs(input):copy(input)
    end
    if self.p > 0 then
-      if self.train then
+      if self.train or self.stochastic_inference and not self.train and not self.v2 then
```

Comment:
hi, just looked at the logic again. isn't this wrong?
if it is evaluate mode, this will never come here because self.v2 will almost always be true.

shouldn't it just be:

if self.train or self.stochastic_inference